systemFlake.nix

```diff
@@ -60,15 +62,25 @@ let
     {
       inherit (selectedNixpkgs) system;
       modules = [
-        ({ pkgs, lib, ... }: {
-          networking.hostName = hostname;
-
-          nixpkgs = {
-            inherit (selectedNixpkgs) overlays config system;
-          };
-
-          system.configurationRevision = lib.mkIf (self ? rev) self.rev;
-          nix.package = lib.mkDefault pkgs.nixUnstable;
+        ({ pkgs, lib, options, ... }: {
+          # 'mkMerge` to separate out each part into its own module
+          _type = "merge";
+          contents = [
```

Comment:
"mkMerge" is necessary to prevent infinite recursion. You can't just merge the modules together.

Reply:
Also this is just a manual `mkMerge` from  nixos module system, if that wasn't clear already.